<?xml version="1.0"?>
<launch>

    <arg name="gazebo" default="false" />
    <arg name="hand_name"/>
    <arg name="hal_lib"/>
    <arg name="gdb" default="false"/> <!-- for debug with gdb -->
    <arg name="matlogger_path" default=""/> 



    <node unless="$(arg gdb)" exec="EEHalExecutor" name="EEHalExecutor" pkg="end_effector" output="screen">
        <param name="hal_library_name" value="$(var hal_lib)"/> 
        <param name="matlogger_path" value="$(var matlogger_path)"/> 
    </node>
    <node if="$(arg gdb)" exec="EEHalExecutor" name="EEHalExecutor" pkg="end_effector" output="screen" launch-prefix="xterm -e gdb --args">
        <param name="hal_library_name" value="$(var hal_lib)"/> 
        <param name="matlogger_path" value="$(var matlogger_path)"/> 
    </node>
    
    <!-- Joint state publisher is necessary for mimic joint, not used if gazebo is used because there is the gazebo_ros_plugin which handles the mimic -->
    <node unless="$(arg gazebo)" name="joint_state_publisher" pkg="joint_state_publisher" exec="joint_state_publisher">
        <!-- Input -->
        <rosparam param="source_list">["/dummyHal/joint_command"] </rosparam>
        <!-- Output -->
        <remap from="joint_states" to="/dummyHal/joint_states" />
        
    </node>
    
<!--    //TODO -->
<!--    
    <include if="$(arg gazebo)" file="$(find rosee_gazebo_plugins)/launch/gazebo.launch"> 
    	  <arg name="hand_name" value="$(arg hand_name)"/>
    </include>   -->

    
</launch>
